vyper/exceptions.py

```diff
@@ -31,68 +32,79 @@ class VyperException(Exception):
     This exception is not raised directly. Other exceptions inherit it in
     order to display source annotations in the error string.
     """
-    def __init__(self, message='Error Message not found.', item=None):
+
+    def __init__(self, message="Error Message not found.", *items):
         """
         Exception initializer.
 
         Arguments
         ---------
         message : str
             Error message to display with the exception.
-        item : VyperNode | tuple, optional
-            Vyper ast node or tuple of (lineno, col_offset) indicating where
-            the exception occured.
+        *items : VyperNode | tuple, optional
+            Vyper ast node(s) indicating where the exception occured. Source
+            annotation is generated in the order the nodes are given. A single
+            tuple of (lineno, col_offset) is also understood to support the old
+            API, but new exceptions should not use this approach.
         """
         self.message = message
         self.lineno = None
         self.col_offset = None
 
-        if isinstance(item, tuple):
-            self.lineno, self.col_offset = item[:2]
-        elif hasattr(item, 'lineno'):
-            self.lineno = item.lineno
-            self.col_offset = item.col_offset
-            self.source_code = item.full_source_code
+        if len(items) == 1 and isinstance(items[0], tuple):
```

Comment:
I think instead of allowing a tuple, you should have to apply it via `*tuple_arg` now, then we can remove this `if-else` case

Reply:
Have to preserve this to keep the old API..  we can phase it out once we've refactored away more of `parser`